dependencies.yaml

```diff
@@ -173,7 +173,6 @@ dependencies:
               cuda: "12.*"
             packages:
               - libnvjpeg-dev
-              - libnvjpeg-static
```

Comment:
Now that the nvJPEG version we need is in the CTK and there are Conda packages for it, we have switched to dynamic linking

Already commented on the relevant build logic making this change above